### PR TITLE
Added all the features

### PR DESCRIPTION
**Pull request by robin-khurana**
Added Mailer, calender view as well bootstrap to the interview creation App
-> Added the mailer for interview update, declined and completed
-> Added the reminder mailer which will deliver mail 30 minutes prior to the interview which will be created when interview status got approved

Added mailcatcher screenshots for mail verification
Screenshot 2020-05-20 at 6 39 54 AM
Screenshot 2020-05-20 at 6 41 02 AM
Screenshot 2020-05-20 at 6 41 30 AM
Screenshot 2020-05-20 at 6 41 14 AM

Done with complete erb Interview creation App for now

### CODE REVIEW

File: test/mailers/previews/interview_mailer_preview.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Preview all emails at http://localhost:3000/rails/mailers/interview_mailer
+class InterviewMailerPreview < ActionMailer::Preview
+  def reminder_interviewee_mail
+    interview = Interview.find(2)
```
Comment (jaindheerain) [inline test/mailers/previews/interview_mailer_preview.rb:4]: ?

File: test/mailers/previews/interview_mailer_preview.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    InterviewMailer.reminder_interviewee_mail(interview)
+  end
+
+  def reminder_interviewer_mail
+    interview = Interview.find(3)
```
Comment (jaindheerain) [inline test/mailers/previews/interview_mailer_preview.rb:9]: ?

File: app/models/interview.rb [hunk 3]
```diff
@@ -32,12 +35,50 @@ def assign_interviewer(interviewer_id)
...
+      ReminderJob.set(wait_until: reminder_time).perform_later(self.id)
+    when'declined'
+      send_declined_mail
+    when'completed'
+      send_completed_mail
+    end
+  end
+
+  def send_updation_mail
+    InterviewMailer.update_interviewer_email(self.id).deliver_now
```
Comment (jaindheerain) [inline app/models/interview.rb:62]: user deliver_later always

File: app/models/interview.rb [hunk 4]
```diff
@@ -32,12 +35,50 @@ def assign_interviewer(interviewer_id)
     UserInterviewRole.create(user_id: interviewer_id, role_id: role.id, interview_id: self.id)
   end
 
+  def send_reminder_mail
+    InterviewMailer.reminder_interviewee_mail(self.id).deliver_now
```
Comment (jaindheerain) [inline app/models/interview.rb:39]: here also

File: app/models/interview.rb [hunk 5]
```diff
@@ -1,5 +1,8 @@
 class Interview < ApplicationRecord
 
+  after_update :after_status_update, if: :saved_change_to_status?
```
Comment (jaindheerain) [inline app/models/interview.rb:3]: if interview time updated, still the old reminders will be sent/

File: test/mailers/previews/interview_mailer_preview.rb (hunk 1 above)
Comment (robin-khurana) [reply test/mailers/previews/interview_mailer_preview.rb:4]: I just did this, just for testing, I should have to handle cases here. and use Interview.first, Sorry for that

**Pull request merged**